src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py

```diff
@@ -482,3 +482,48 @@ def test_kerncallarglist_scalar_literal(fortran_writer):
         create_arg_list.scalar(args[3])
     assert ("Unexpected literal expression 'invalid' in scalar() when "
             "processing kernel 'testkern_qr_code'" in str(err.value))
+
+
+def test_indirect_dofmap_apply(fortran_writer):
+    '''Test the indirect dofmap apply function.
+    '''
+    psy, _ = get_invoke("20.1.2_cma_apply_disc.f90", TEST_API,
+                        dist_mem=False, idx=0)
+
+    schedule = psy.invokes.invoke_list[0].schedule
+    create_arg_list = KernCallArgList(schedule.kernels()[0])
+    create_arg_list.generate()
+    assert (create_arg_list._arglist == [
+        'cell', 'ncell_2d', 'field_a_proxy%data', 'field_b_proxy%data',
+        'cma_op1_matrix', 'cma_op1_nrow', 'cma_op1_ncol', 'cma_op1_bandwidth',
+        'cma_op1_alpha', 'cma_op1_beta', 'cma_op1_gamma_m', 'cma_op1_gamma_p',
+        'ndf_adspc1_field_a', 'undf_adspc1_field_a',
+        'map_adspc1_field_a(:,cell)', 'cma_indirection_map_adspc1_field_a',
+        'ndf_aspc1_field_b', 'undf_aspc1_field_b', 'map_aspc1_field_b(:,cell)',
+        'cma_indirection_map_aspc1_field_b'])
+
```

Comment:
Could you add a check that suitable symbols have been added to the correct table?

Reply:
I have opened #2022 for that ... for whatever reason I completely forgot to test ANY symbol if they have the correct type :( I've fixed this up for this and the next diff below, but in #2022 I will implement a more readable solution (i.e. using names instead of index number) for all arguments (including actually testing all arguments in the corresponding test). But there needs to be added a lot, too much for this small PR. There is also a #TODO for this, since atm we can't easily create a 1d-array with precision r_solver, which would simplify tests further.

Reply:
Thanks. Could we use the LFRic `domain/lfric/psyir.py` for this instead (in #2022) - it has various types defined in it already?

Reply:
I have replaced the scalar tests, that was easy. With arrays it wasn't that straight forward, so I'll left this for later.

Reply:
I checked it in more detail: reason for the issue with arrays is that the LFRicSymbolTable creates an ArrayType, while `psyir` has a `RealFieldDataDataType` (sic), and that `Datatype.__eq__` requires that the types are the same (`is`). That could be fixed, but certainly too much to do it here.

Reply:
Thanks for looking Joerg. I didn't intend for you to try and do that here :-)